router/src/main/resources/interceptor/lib/utils.lua

```diff
@@ -86,3 +90,43 @@ function table.shallow_copy(src, des)
         des[k] = v
     end
 end
+
+--- log body and headers retrieved from the handle
+---@param handle table
+---@param log_message string
+function wire_log_body(handle, log_message, wire_log_enabled)
+    if wire_log_enabled then 
+        local log_output = "\n" .. "[wirelog]" .. log_message .. "\n"
+        if handle:body() then
+            handle:logInfo(log_output .. handle:body():getBytes(0, handle:body():length()) .. log_output)
```

Comment:
What would be the behavior if the payload is chunked? `transfer-encoding : chunked`

Reply:
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter as per docs, I think it waits till the whole payload is received.

Reply:
yes it waits for the whole body, but there is a configurable limit, default to 1MB
`bodyChunks` method read chunk by chunk and it is not buffering
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener.proto#:~:text=in%20this%20field.-,per_connection_buffer_limit_bytes,-(UInt32Value)%20Soft

Reply:
request body could be logged with `transfer-encoding : chunked` header is set